### PR TITLE
Bucket Sort in Go

### LINKED ISSUE #2124: Bucket Sort
- [x] C
- [x] C++
- [ ] C# - @luv4bytes
- [ ] CoffeeScript
- [ ] Dart
- [x] Go - @m-code12
- [x] Java
- [x] JS - @abhu-A-J
- [x] Kotlin - @shubhammantri1
- [ ] Php - @srimani-programmer
- [x] Python
- [x] Readme
- [x] Ruby - @m-code12
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by m-code12**
Fixes #2124 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Bucket Sort in go

#### Languages Used:
- Go

#### Files Added:
- Bucket_Sort.go

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by m-code12** [comment]
I have implemented Bucket Sort in Go following all the required coding conventions. Please review and merge @jainaman224 @singh-shreya6

File: Bucket_Sort/Bucket_Sort.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package main
```
Comment (singh-shreya6) [inline Bucket_Sort/Bucket_Sort.go:1]: Please add a brief problem description

File: Bucket_Sort/Bucket_Sort.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+)
+
+func insertionSort(array []float64) {
+    for i := 0; i < len(array); i++ {
+        temp := array[i]
+        j := i - 1
+        for ; j >= 0 && array[j] > temp; j-- {
+            array[j+1] = array[j]
+        }
+        array[j+1] = temp
```
Comment (singh-shreya6) [inline Bucket_Sort/Bucket_Sort.go:16]: Space around +

File: Bucket_Sort/Bucket_Sort.go (hunk 1 above)
Comment (singh-shreya6) [inline Bucket_Sort/Bucket_Sort.go:1]: Please add a brief problem description

File: Bucket_Sort/Bucket_Sort.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    "os"
+    "strconv"
+)
+
+func insertionSort(array []float64) {
+    for i := 0; i < len(array); i++ {
+        temp := array[i]
+        j := i - 1
+        for ; j >= 0 && array[j] > temp; j-- {
+            array[j+1] = array[j]
```
Comment (singh-shreya6) [inline Bucket_Sort/Bucket_Sort.go:14]: Space around +

File: Bucket_Sort/Bucket_Sort.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    for _, n := range array {
+        if n < min {
+            min = n
+        }
+        if n > max {
+            max = n
+        }
+    }
+	
+    nBuckets := int(max-min)/bucketSize + 1
```
Comment (singh-shreya6) [inline Bucket_Sort/Bucket_Sort.go:31]: Space around /

File: Bucket_Sort/Bucket_Sort.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    var x int
+    fmt.Printf("Enter the size of the array : ")
+    fmt.Scan(&x)
+    fmt.Printf("Enter the elements of the array : ")
+    a := make([]float64, x)
+    for i := 0; i < x; i++ {
+        fmt.Scan(&a[i])
+    }
+ 
+    array :=a    // A copy of the array 'a' is assigned to 'array'
```
Comment (singh-shreya6) [inline Bucket_Sort/Bucket_Sort.go:62]: Soace after :=

**Comment by m-code12** [comment]
I have made the required changes . Please review @singh-shreya6

**Pull request merged**